src/Symfony/Component/Serializer/Encoder/DecoderInterface.php

```diff
@@ -30,9 +30,11 @@ interface DecoderInterface
      * are encouraged to document which formats they support in a non-inherited
      * phpdoc comment.
      *
+     * @return mixed
+     *
      * @throws UnexpectedValueException
      */
-    public function decode(string $data, string $format, array $context = []): mixed;
+    public function decode(string $data, string $format, array $context = []);
```

Comment:
Okay, and the return type on `supportsDecoding()` is not a problem?

Reply:
Reverted also while merging

Reply:
Nope https://github.com/api-platform/core/blob/main/src/Serializer/JsonEncoder.php#L58.  Problematics declarations are mostly PHP8+ types (mixed, static, union)

Reply:
I just see that we also removed some `bool` return types in #44331. 
@dunglas Could you please double check if we didn't remove more type declarations than necessary?

Reply:
It wouldn't make sense to add a return on half the interface if we think with FC/BC in mind. APIP is not the only target :)

Reply:
Well, if the target is PHP 7.2+ projects, then any type supported in PHP 7.2 is fine.
If the target is PHP 7.1 projects targetting Symfony 4.4 to 6, then any return type is an issue.